web/src/pages/Settings/Settings.js

```diff
@@ -0,0 +1,153 @@
+import React, {useCallback, useEffect, useRef, useState} from 'react'
+import './Settings.css'
+import ReactCrop from 'react-image-crop';
+import 'react-image-crop/dist/ReactCrop.css';
+
+const FormData = require('form-data');
+
+export default (props) => {
+  const [image, updateImage] = useState()
+  const imgRef = useRef(null);
+  const [loading, updateLoading] = useState(true)
+  const [modalOpened, updateModalOpened] = useState(false)
+  const [me, updateMe] = useState("")
+  const [crop, setCrop] = useState(
+    {
+      unit: '%',
+      aspect: 1,
+      x: 0,
+      y: 0,
+      width: 100,
+    });
+  const [upImage, updateUpImage] = useState()
+
+  useEffect(async () => {
+    const meProfile = await props.api.getMe()
+    updateMe(meProfile)
+    updateImage(meProfile.avatar_url)
+    updateLoading(false)
+  }, [])
+
+  const changeAvatarOnClick = (event) => {
+    if (event.target.files && event.target.files[0]) {
+      let img = event.target.files[0];
+      updateUpImage(URL.createObjectURL(img))
+      updateModalOpened(true)
+    }
+  }
+
+  /**
+   * @param {HTMLImageElement} image - Image File Object
+   * @param {Object} crop - crop Object
+   * @param {String} fileName - Name of the returned file in Promise
+   */
+  function getCroppedImg(image, crop, fileName) {
+    const canvas = document.createElement('canvas');
+    const scaleX = image.naturalWidth / image.width;
+    const scaleY = image.naturalHeight / image.height;
+    canvas.width = crop.width;
+    canvas.height = crop.height;
+    const ctx = canvas.getContext('2d');
+
+    ctx.drawImage(
+      image,
+      crop.x * scaleX,
+      crop.y * scaleY,
+      crop.width * scaleX,
+      crop.height * scaleY,
+      0,
+      0,
+      crop.width,
+      crop.height,
+    );
+
+    // As a blob
+    return new Promise((resolve, reject) => {
+      canvas.toBlob(blob => {
+        blob.name = fileName;
+        resolve(blob);
+      }, 'image/jpeg', 1);
```

Comment:
What if the image is a png

Reply:
I'm pretty sure you can upload png, I enabled all image type in `Settings.js` line 94. Do you want to limit image type down to jpg/png only?